universal-login-sdk/src/api/SdkSigner.ts

```diff
@@ -2,19 +2,20 @@ import UniversalLoginSDK from './sdk';
 import {ethers, Wallet, utils, providers} from 'ethers';
 import {ensureNotFalsy, DEFAULT_GAS_LIMIT} from '@universal-login/commons';
 import {TransactionHashNotFound} from '../core/utils/errors';
+import {DeployedWallet} from './wallet/DeployedWallet';
 
 export class SdkSigner extends ethers.Signer {
   private wallet: Wallet;
   provider: providers.Provider;
 
   constructor(
     private sdk: UniversalLoginSDK,
+    private deployedWallet: DeployedWallet,
     public contractAddress: string,
-    privateKey: string,
   ) {
     super();
     this.provider = sdk.provider;
```

Comment:
You can get the provider from `deployedWallet.sdk.provider` and remove the constructor param